substrate/primitives/api/proc-macro/src/utils.rs

```diff
@@ -330,4 +391,32 @@ mod tests {
 		assert_eq!(cfg_std, filtered[0]);
 		assert_eq!(cfg_benchmarks, filtered[1]);
 	}
+
+	#[test]
+	fn check_deprecated_attr() {
+		const FIRST: &'static str = "hello";
+		const SECOND: &'static str = "WORLD";
+
+		let simple: Attribute = parse_quote!(#[deprecated]);
+		let simple_path: Attribute = parse_quote!(#[deprecated = #FIRST]);
+		let meta_list: Attribute = parse_quote!(#[deprecated(note = #FIRST)]);
+		let meta_list_with_since: Attribute =
+			parse_quote!(#[deprecated(note = #FIRST, since = #SECOND)]);
```

Comment:
Would be interesting to see how this behaves with extra parameters

Reply:
it will just ignore them

Reply:
Do you think we can turn that into a compile error? And specify inside the message the forms we support?

Reply:
it's a compiler error already, that's one of the reason why i decided to use rust's deprecated. But the parsing now doesnt fail if attribute is malformed(not sure how useful it is tho)